src/tests/ftest/util/dmg_utils.py

```diff
@@ -897,6 +897,29 @@ def system_stop(self, force=False, ranks=None):
                 data[rank] = info[1].strip()
         return data
 
+    def system_leader_query(self):
+        """Query the system MS leader.
+
+        Raises:
+            CommandFailure: if the dmg system stop command fails.
+
+        Returns:
+            dict: a dictionary of host ranks and their unique states.
+
+        """
+        self._get_result(("system", "leader-query"))
+
+        # Current Leader: 10.7.1.10:10001
+        #    Replica Set: 10.7.1.10:10001, 10.7.1.68:10001, 10.7.1.9:10001
+        data = {}
+        match = re.findall(
+            r"(?:Current Leader|Replica Set):\s+([0-9.:, ]+)",
+            self.result.stdout)
```

Comment:
I really think that we shouldn't be writing new tests that use the stdout scraper, and instead should be using the --json output. It will be more stable and less prone to breaking tests when there are cosmetic updates to the output for humans.